ov/id_pools_ipv4_subnets.go

```diff
@@ -0,0 +1,298 @@
+	ResourceUri      utils.Nstring `json:"resourceUri,omitempty"`
+}
+
+type SubnetList struct {
+	Total       int           `json:"total,omitempty"`       // "total": 1,
+	Count       int           `json:"count,omitempty"`       // "count": 1,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	URI         utils.Nstring `json:"uri,omitempty"`         // "uri": "/rest/id-pools/ipv4/subnets?filter=networkId='10.10'"
+	Members     []Ipv4Subnet  `json:"members,omitempty"`     // "members":[]
+}
+
+type SubnetAllocatorList struct {
+	Count  int             `json:"count,omitempty"`
+	ETAG   string          `json:"eTag,omitempty"`
+	Valid  bool            `json:"valid,omitempty"`
+	IdList []utils.Nstring `json:"idList,omitempty"`
+}
+
+type SubnetCollectorList struct {
+	ETAG   string          `json:"eTag,omitempty"`
+	IdList []utils.Nstring `json:"idList,omitempty"`
+}
+
+func (c *OVClient) GetIPv4SubnetbyId(id string) (Ipv4Subnet, error) {
+	var (
+		uri        = "/rest/id-pools/ipv4/subnets/"
+		ipv4Subnet Ipv4Subnet
+	)
+
+	uri = uri + id
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return ipv4Subnet, err
+	}
+
+	log.Debugf("Get Ipv4 Subnets %s", data)
+	if err := json.Unmarshal([]byte(data), &ipv4Subnet); err != nil {
+		return ipv4Subnet, err
+	}
+	return ipv4Subnet, nil
+}
+
+func (c *OVClient) GetSubnetByNetworkId(nwId string) (Ipv4Subnet, error) {
+	var (
+		subnet Ipv4Subnet
+	)
+	subnets, err := c.GetIPv4Subnets("", "", fmt.Sprintf("networkId='%s'", nwId), "")
+	if subnets.Total > 0 {
+		return subnets.Members[0], err
+	} else {
+		return subnet, err
+	}
+}
+
+func (c *OVClient) GetIPv4Subnets(start string, count string, filter string, sort string) (SubnetList, error) {
+	var (
+		uri     = "/rest/id-pools/ipv4/subnets/"
+		q       map[string]interface{}
+		subnets SubnetList
+	)
+	q = make(map[string]interface{})
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return subnets, err
+	}
+
+	log.Debugf("Get All Subnets", data)
+	if err := json.Unmarshal([]byte(data), &subnets); err != nil {
+		return subnets, err
+	}
+	return subnets, nil
+}
+
+func (c *OVClient) CreateIPv4Subnet(subnet Ipv4Subnet) error {
+	var (
+		uri = "/rest/id-pools/ipv4/subnets/"
+		t   = (&Task{}).NewProfileTask(c)
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t.ResetTask()
```

Comment:
If youre using Task, use t.wait()
I am not sure if this is being completely utilized here.